src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -447,7 +447,11 @@ public function isAbsolutePath($file)
     private function toIterator($files)
     {
         if (!$files instanceof \Traversable) {
-            $files = new \ArrayObject(is_array($files) ? $files : array($files));
+            /** no direct array casting (objects casted to properties array) */
+            if ((array) $files !== $files) {
+                $files = array($files);
+            }
```

Comment:
I'm not aim to confuse maintainers, so can just revert 'if ((array) $files !== $files)' back to original.

So, I shall revert, right?